packages/accounts/src/light-account/account.ts

```diff
@@ -65,13 +76,24 @@ export type CreateLightAccountParams<
   factoryAddress?: Address;
   initCode?: Hex;
   version?: LightAccountVersion;
-  entryPoint?: EntryPointDef<TEntryPointVersion, Chain>;
-};
+} & (OneOf<TEntryPointVersion, TEntryPointVersion> extends never
+    ? { entryPoint: never; version?: never }
+    : TEntryPointVersion extends "0.6.0"
+    ? {
+        entryPoint?: EntryPointDef<TEntryPointVersion, Chain>;
+        version?: Extract<LightAccountVersion, `v1${string}`>;
+      }
+    : TEntryPointVersion extends "0.7.0"
+    ? {
+        entryPoint: EntryPointDef<TEntryPointVersion, Chain>;
+        version: Exclude<LightAccountVersion, `v1${string}`>;
+      }
+    : { entryPoint?: never; version?: never });
 
 export async function createLightAccount<
```

Comment:
![Screenshot 2024-04-03 at 11 37 50 PM](https://github.com/alchemyplatform/aa-sdk/assets/3278577/0830d2df-4c98-472f-9e2c-53a91a049301)

Not sure why, but if I have the same approach for simple account on aa-core, typescript errors saying the signatures are not compatible for these overloaded functions.

Reply:
huh it turns out to be zod typescript issue.